### PR TITLE
Update for 2017 mpl ideas

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: 2017/ideas-list-matplotlib.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+| **Intensity** | **Involves**  | **Mentors** |
+| ------------- | --------------|------------ |
+| Advanced | Python, Constraint Solving | [@mdboom][] [@tacaswell][] |
+
+### Technical details
+
+This work will likely be based on an existing constraint-solving
+library, such as
+[Cassowary](http://constraints.cs.washington.edu/cassowary/).  Likely
```
Comment (QuLogic) [inline 2017/ideas-list-matplotlib.md]: https

File: 2017/ideas-list-matplotlib.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+this would be implemented in two phases: (1) the implementation of
+existing layout abilities in terms of constraints, accessible through
+a new API, and (2) exposing this new implementation through the
+existing APIs.
+
+### Open Source Development Experience
+
+This project involves experience with Python and and understanding of
```
Comment (QuLogic) [inline 2017/ideas-list-matplotlib.md]: Double 'and'.

File: 2017/ideas-list-matplotlib.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+### Technical details
+
+Beside just writing, this task will involve working closely with our
+Sphinx-based documentation production system, possibly writing and
+adapting extensions.  Writing tools and scripts in Python to automate
+repetitive tasks will also be required.
+
+### Open Source Development Experience
+
+A contributor to this project will have to have broad experience using
```
Comment (QuLogic) [inline 2017/ideas-list-matplotlib.md]: will have to -> should

File: 2017/ideas-list-matplotlib.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+<a name="text"></a>
+## International text handling
+
+### Abstract
+
+Matplotlib's existing text layout engine is appropriate only for
+standard English-like left-to-right languages.  In order to open
+Matplotlib usage up to a broader international audience, it would be
+nice to wrap an existing internationalized text layout engine, such as
+[Pango](http://www.pango.org/)/[Harfbuzz](https://github.com/behdad/harfbuzz).
```
Comment (QuLogic) [inline 2017/ideas-list-matplotlib.md]: http://harfbuzz.org/
The GitHub repo is not canonical.

File: 2017/ideas-list-matplotlib.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+important one is detecting when plotting categorical data
+(i.e. enumerations) and updating the tick labels accordingly.
+
+| **Intensity** | **Involves**  | **Mentors** |
+| ------------- | --------------|------------ |
+| Intermediate | Python, Pandas, Databases, Data science | [@tacaswell][] |
+
+### Technical details
+
+More broadly, this task will involve designed new user-friendly APIs
```
Comment (QuLogic) [inline 2017/ideas-list-matplotlib.md]: designed -> designing

**Comment by kain88-de** [comment]
@tacaswell Good to hear that you want to take part again. How did you handle the projects last year with PR's? Did you first finish it and merged at the end of the organization application period or merged every once and a while?

**Comment by tacaswell** [comment]
I do not recall :disappointed: .  I think either way will work.

**Comment by kain88-de** [comment]
OK Then I'll merge this once you fixed the remarks of @QuLogic. Afterwards you can create a new PR and I'll merge it every few days when there are larger changes.

File: 2017/ideas-list-matplotlib.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+existing APIs.
+
+### Open Source Development Experience
+
+This project involves experience with Python and and understanding of
+linear constraint solvers.  Computer graphics concepts, particularly
+text layout, would also be an asset.
+
+<a name="documentation"></a>
+## Documentation improvements
```
Comment (rgaiacs) [inline 2017/ideas-list-matplotlib.md]: This is out of Google of Summer Code scope. Sorry. The students **must** to write code. But this project is suitable for Google Code-in.

Comment (tacaswell) [inline 2017/ideas-list-matplotlib.md]: Would writing sphinx extensions be in scope?

Comment (rgaiacs) [inline 2017/ideas-list-matplotlib.md]: Yes. If the scope of the project is a sphinx extension it is valid and part of the student time could be to improve matplotlib documentation as a case **but** I recommend that 60% of student time be dedicate to write the sphinx extension.

Comment (kain88-de) [inline 2017/ideas-list-matplotlib.md]: This idea needs some more work in defining the scope to be viable for GSoC. I would also recommend that you have yourself a strong idea how the documentation should be organized and know where the sphinx extensions aren't doing what you need or where repetition can be automated. 

My gut feeling is though that this project is better suited to distribute over small Code-In tasks. We can ask later this year if organizations would be interested in taking  part in Code-In with NumFOCUS as an umbrella.

File: 2016/ideas-list-matplotlib.md [hunk 7]
```diff
@@ -30,13 +21,14 @@ layout difficulties such as twinned axes and colorbars.
...
 | ------------- | --------------|------------ |
-| Advanced | Python, Constraint Solving | [@mdboom][] [@tacaswell][] |
+| Advanced | Python, Constraint Solving | [@tacaswell][] |
 
 ### Technical details
 
 This work will likely be based on an existing constraint-solving
 library, such as
-[Cassowary](http://constraints.cs.washington.edu/cassowary/).  Likely
+[Cassowary](http://constraints.cs.washington.edu/cassowary/) as implemneted by
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: implemented

File: 2016/ideas-list-matplotlib.md [hunk 8]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
+| Intermediate  | Python, | [@tacaswell][] |
+
+
+### Technical details
 
-1. Reorganizing the matplotlib [gallery](http://matplotlib.org/gallery.html) into albums
+One approach at this would be add some fast-paths through the existing code.
 
-1. Moving the [examples](http://matplotlib.org/gallery.html) into albums
+Extending the current interpolation system to inculde dedicated (external) resampling code, for
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: include

File: 2016/ideas-list-matplotlib.md [hunk 9]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
+example datashader.  See [bokeh/datashader#200](https://github.com/bokeh/datashader/pull/200) for
+a proof of concept.
 
+Folding something
+like
+[mpl-modest-image](https://github.com/ChrisBeaumont/mpl-modest-image)
+directly into matplotlib would alo be in scope for this project.
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: also

File: 2016/ideas-list-matplotlib.md [hunk 10]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
+
+### Open Source Development Experince
+
+This task will be majority in python and requires a working knowledge
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: Python

File: 2016/ideas-list-matplotlib.md [hunk 11]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
+of numpy.  Domain experiance with large images is a plus.
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: NumPy; experience

File: 2016/ideas-list-matplotlib.md [hunk 12]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
+
+This work may require the ability to read and understand templated c++
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: C++

File: 2016/ideas-list-matplotlib.md [hunk 13]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
+and Matplotlib's binding of Agg to python.
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: Python

File: 2016/ideas-list-matplotlib.md [hunk 14]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
+
+This work may require API design (for which domain expertise will be beneficial) and
+interaction with external libaries.
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: libraries

File: 2016/ideas-list-matplotlib.md [hunk 15]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
+
+
+## 2D color maps
+### Abstract
+All of the color mapping in Matplotlib is currently derived from
+`ScalerMappable` which as the same suggests maps scalers from `R^1 ->
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: name

File: 2016/ideas-list-matplotlib.md [hunk 16]
```diff
@@ -49,56 +41,89 @@ linear constraint solvers.  Computer graphics concepts, particularly
...
-repetitive tasks will also be required.
+In addition, a 2D color bar would need to be implemented and suitable
+color maps will need to be developed.
 
 ### Open Source Development Experience
 
-A contributor to this project will have to have broad experience using
-matplotlib and Python.  A deep understanding of Sphinx, including writing extensions,
-would also be a major asset.
+Competence with python and OO design is required.  Experience with the
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: Python

File: 2016/ideas-list-matplotlib.md [hunk 17]
```diff
@@ -202,41 +203,51 @@ This work should be 100% in Python.  Understanding of duck-typing and
...
+Writing tools and scripts in Python to automate repetitive tasks will
+also be required.
 
 ### Open Source Development Experience
 
-This work should be mostly in Python with some Javascript and
-understanding of browser technologies.  Understanding of multiple GUI
-frameworks a must.  Access to multiple operating systems (as many as
-possible of Linux, Mac, Windows, iOS, and Android), is very helpful.
+A contributor to this project will have to have broad experience using
```
Comment (QuLogic) [inline 2016/ideas-list-matplotlib.md]: will have to -> should

**Comment by tacaswell** [comment]
Oh goodness, I have now edited both the 2016 and 2017 file...

**Comment by kain88-de** [comment]
Thanks looks great

**Pull request merged**